packages/twenty-server/src/engine/workspace-manager/workspace-manager.service.ts

```diff
@@ -168,4 +186,35 @@ export class WorkspaceManagerService {
     // Delete schema
     await this.workspaceDataSourceService.deleteWorkspaceDBSchema(workspaceId);
   }
+
+  private async initPermissions(workspaceId: string) {
+    await this.permissionsService.createAdminRole({
+      workspaceId,
+    });
+
+    const userWorkspace = await this.userWorkspaceRepository.find({
+      where: {
+        workspaceId,
+      },
+    });
+
+    if (isEmpty(userWorkspace)) {
+      throw new PermissionsException(
+        'User workspace not found',
+        PermissionsExceptionCode.USER_WORKSPACE_NOT_FOUND,
+      );
+    }
+
+    if (userWorkspace.length > 1) {
+      throw new PermissionsException(
+        'Multiple user workspaces found, cannot tell which one should be admin',
+        PermissionsExceptionCode.TOO_MANY_ADMIN_CANDIDATES,
+      );
+    }
+
+    await this.permissionsService.assignAdminRoleToUserWorkspace(
```

Comment:
We could have a generic .assignRoleToUserWorkspace and pass the id which we got directly as a response to our `createAdminRole` function